ch_pipeline/analysis/flagging.py

```diff
@@ -139,6 +142,48 @@ def process(self, data) -> Union[containers.RFIMask, dcontainers.RFIMask]:
         return out
 
 
+class RFIStaticMask(task.SingleTask):
+    """Get the static mask for the time period covered by the data.
+
+    This is the same static mask used in :class:`RFIFilter` and
+    :class:`RFISensitivityMask`.
+
+    Parameters
+    ----------
+    invert : bool, optional
+        Whether to invert the mask. Defaults to False.
+    """
+
+    invert = config.Property(proptype=bool, default=False)
+
+    def process(self, data) -> Union[dcontainers.RFIMask, dcontainers.SiderealRFIMask]:
```

Comment:
```suggestion
    @overload
    def process(self, data: dcontainers.TimeStream) -> dcontainers.RFIMask:
        ...
    @overload
    def process(self, data: dcontainers.SiderealStream) -> dcontainers.SiderealRFIMask:
        ...
    def process(self, data):
```
For my own interest I was trying to figure out how to type this. It's a bit problematic because what you really want are intersection types (apparently), e.g. data must be a `SiderealContainer` and a `FreqContainer` but with no other constraints, but they don't really exist. I think the most likely things to apply this too are TimeStreams and SiderealStream's so this is about the best that can be done.

Reply:
Seems like there was some discussion about it [here](https://github.com/python/typing/issues/18) but it doesn't look like there are any plans to introduce an Intersection/Any type in the way that we would want here. So I agree that using `overload` is probably the best option at the moment